samples/snippets/data_client/data_client_snippets_async.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python
+
+# Copyright 2024, Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+async def write_simple(table):
+    # [START bigtable_async_write_simple]
+    from google.cloud.bigtable.data import BigtableDataClientAsync
+    from google.cloud.bigtable.data import SetCell
+
+    async def write_simple(project_id, instance_id, table_id):
+        async with BigtableDataClientAsync(project=project_id) as client:
+            async with client.get_table(instance_id, table_id) as table:
+                family_id = "stats_summary"
+                row_key = b"phone#4c410523#20190501"
+
+                cell_mutation = SetCell(family_id, "connected_cell", 1)
+                wifi_mutation = SetCell(family_id, "connected_wifi", 1)
+                os_mutation = SetCell(family_id, "os_build", "PQ2A.190405.003")
+
+                await table.mutate_row(row_key, cell_mutation)
+                await table.mutate_row(row_key, wifi_mutation)
+                await table.mutate_row(row_key, os_mutation)
+
+    # [END bigtable_async_write_simple]
+    await write_simple(table.client.project, table.instance_id, table.table_id)
+
+
+async def write_batch(table):
+    # [START bigtable_async_writes_batch]
+    from google.cloud.bigtable.data import BigtableDataClientAsync
+    from google.cloud.bigtable.data.mutations import SetCell
+    from google.cloud.bigtable.data.mutations import RowMutationEntry
+
+    async def write_batch(project_id, instance_id, table_id):
+        async with BigtableDataClientAsync(project=project_id) as client:
+            async with client.get_table(instance_id, table_id) as table:
+                family_id = "stats_summary"
+
+                async with table.mutations_batcher(
+                    flush_limit_mutation_count=2, flush_limit_bytes=1024
```

Comment:
should we add a comment here saying the default is 100 and is good for most use cases, setting it to 2 just for the example purpose?
Or we don't change the defaults here, and just manually flush in the end. And I would also add some comments on why manual flush is needed, something like "flush will be called automatically when a batch is full, the default batch size is 100".

Reply:
Ok, maybe I'll remove the arguments here if that's confusing. A manual flush isn't needed since we are using a `async with` statement, it will flush and close everything for us at the end of this block

Let me know if you think any extra comments are needed to clarify anything